### DISCUSSION TITLE
Some questions, ideas, and possible issues with World Info

### DISCUSSION THREAD

**Discussion by FaceDeer** [LostRuins/koboldcpp#515] [General]
Please pardon if any of this is off base, but I'm fairly new to fiddling with KoboldAI Lite and with these sorts of "chat oriented" LLMs in general and I need to start somewhere. :)

I've been having some fun doing worldbuilding exercises and when I saw the "World Info" feature it looked super awesome for this kind of application. However, making good use of it is proving to be a little difficult. It seems like ideally it'd be something that could be saved and loaded like a character is, as a sort of "setting card", but the existing UI is pretty inconvenient to work with. It takes a bunch of clicks to get to adding a new item (first click "W Info", then click a small "+" button, then click the tag field to start typing) and as far as I can see there's no way to save the world info as a separate thing from the session as a whole. One can go into the saved session .json file and manually trim out everything but the world info, but hand-editing json files is pretty technical.

I went digging into the source next, and I found that [klite.embed has support for loading world info in its load_tavern_ob function](https://github.com/LostRuins/koboldcpp/blob/7a8c0df2e51a43a3803255f3477112e60da41c81/klite.embd#L4718). It even has a code path for loading a "world info only card". That was looking promising, so I went in search of a world info card example out on the internet to experiment with. I found [this one](https://gist.github.com/Cohee1207/a74a7adf074d4989fb9dea7dd876a04a), but it didn't work out of the box. I did eventually get it to work by splicing it into an existing character card I had, turning the "entries" object into an array and wrapping the whole thing in a "character_book" object. I then had a character that when loaded would add a bunch of world info along with the rest. But again, that was a bunch of hand-editing of json files. And I wasn't able to get the world info to load by itself, only as part of a character. So the "//check if it's a world info only card, if so, do not restart game" comment in klite.embed doesn't seem to apply.

Is there some workflow that I'm missing here? Granted, I only found that one example of a world info card and it's 7 months old, so perhaps there's a new format available.

My "dream feature" as far as world info goes would be a button similar to the "AutoGenerate Memory" one that leverages the LLM to read through its context and generate a world info entry for the tag that I give it on its own, that way as I'm wandering around exploring aspects of a setting I could very easily add interesting stuff to the list as I go. Ideally with a way to save a world-info-only card, but I could pry it out of the scenario save if needed. But I don't know how much use this feature would see so if it's a lot of effort I can understand it being way down on the priority list.

Should any of this stuff be filed as issues, perhaps?

**Comment by LostRuins** [comment]
You can choose to only import world info from any koboldai save file. This is done by enabling "Show Advanced Load" in settings. Then everything else will not be overwritten.

**Reply by FaceDeer** [reply]
Ah, there it is. Okay, this is workable, thanks! Not the most intuitive thing to find, but this is all very new tech so I can bear that now that I know how to do it.
